src/models.rs

```diff
@@ -93,3 +91,68 @@ pub struct AuthToken {
 }
 
 pub const AUTH_TOKEN_TTL: i64 = 30 * 24 * 60 * 60; // 30 days
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use establish_connection;
+    use schema::users;
+
+    fn conn_with_a_user() -> PgConnection {
```

Comment:
Would it make sense to separate establing db connection logic and factorying users logic? A module like `UserFactory` would do I suppose.

Reply:
This function providing a connection with test_transaction and some data already inserted in the database is inspired by diesel tests ([declaration](https://github.com/diesel-rs/diesel/blob/73778af0827cb066bf1efdb348f380da14c916d4/diesel_tests/tests/schema.rs#L249) and [usage](https://github.com/diesel-rs/diesel/blob/ae706bf157e3d2d5d9639de95d30a23bf27e6ef8/diesel_tests/tests/update.rs#L5)).

By factorying you mean creating a user in the database? Then a data fixtures module in a test helpers module would make sense yes (like in the diesel code I linked).

Reply:
> This function providing a connection with test_transaction and some data already inserted in the database is inspired by diesel tests (declaration and usage).

Yeah it looks perfectly fine the way it is now. My comment was merely about the one below 👇.

> By factorying you mean creating a user in the database? Then a data fixtures module in a test helpers module would make sense yes (like in the diesel code I linked).

Exactly, so we can reuse these factories/fixtures in the future as well (I’m pretty sure we will).

Sorry for the confusion it might cause.